### PR TITLE
Reader: Support an `at` param to set where a stream starts.

### LINKED ISSUE #2574: Reader: Support starting at a date in all streams
Mostly for debugging, it would be really nice to be able to start a stream at a certain date. I think this would work by passing a querystring argument – `at` perhaps – that the stream picks up and starts with. So maybe: 
  http://wordpress.com/?at=2016-01-18T00:00:00-05:00
and
  http://wordpress.com/tag/photo?at=2016-01-10T00:00:00 (UTC)

### PR DESCRIPTION
**Pull request by blowery**
This patch teaches all of the reader streams to start at a certain date. This helps when debugging streams that have a lot of volume, like tag streams, by allowing you to start at a particular date.

The at param will take any ISO-8601 compliant date, with timezone, and translate it to a UTC-based date before passing it down to the API.

Fixes #2574

### CODE REVIEW

**Comment by blowery** [comment]
To test, hit http://calypso.localhost:3000/?at=2016-01-01T00:00:00-05:00 and any other steams you'd like to test with the `?at` param specified.

File: client/reader/controller.js [hunk 1]
```diff
@@ -59,8 +60,14 @@ function removeFullPostDialog() {
 	__fullPostInstance = null;
 }
 
-function ensureStoreLoading( store ) {
+function ensureStoreLoading( store, context ) {
```
Comment (bluefuton) [inline client/reader/controller.js]: Do we need to hand in the entire `context` here, or would `context.query` be enough?

Comment (blowery) [inline client/reader/controller.js]: i didn't want to break apart the context, but yeah. could even just pass the start date directly, but then we have to have the code to pull things off the context everywhere.

**Comment by bluefuton** [comment]
Works a treat in following stream, team stream, My Likes, tag stream, list stream and when viewing an individual blog or feed stream :+1:

Doesn't work for `/discover` shortcut URL, possibly because the controller doesn't provide `context`?

Doesn't appear to like positive timezone offsets like:

http://calypso.localhost:3000/?at=2016-01-01T00:00:00+05:00

**Comment by blowery** [comment]
> Doesn't appear to like positive timezone offsets like:
> http://calypso.localhost:3000/?at=2016-01-01T00:00:00+05:00

That's because `+` is a way to encode a space. :) It has to be http://calypso.localhost:3000/?at=2016-01-01T00:00:00%2B05:00

**Comment by blowery** [comment]
> Doesn't work for /discover shortcut URL, possibly because the controller doesn't provide context?

Huh. http://calypso.localhost:3000/discover redirects me to discover.wordpress.com, but https://wordpress.com/discover shows me the stream. How odd.

**Comment by blowery** [comment]
Fixed the redirect with discover, now `at` seems to work just fine for the discover stream. It was already being passed the context and calling `ensureStoreLoading` with it.

File: client/reader/controller.js (hunk 1 above)
Comment (bluefuton) [inline client/reader/controller.js]: :+1:

**Comment by bluefuton** [comment]
Confirmed that Discover stream `at` now works, as do + timezones when properly URL encoded ;)

**Pull request merged**